mlkem/reduce.c

```diff
@@ -87,6 +88,24 @@ int16_t montgomery_reduce(int32_t a) {
   return (int16_t)r;
 }
 
+int16_t montgomery_reduce(int32_t a) {
+  SCALAR_BOUND(a, 2 * MLKEM_Q * 32768, "montgomery_reduce input");
+
+  int16_t res = montgomery_reduce_generic(a);
+
+  SCALAR_BOUND(res, (3 * (MLKEM_Q + 1)) / 2, "montgomery_reduce output");
+  return res;
+}
+
+int16_t fqmul(int16_t a, int16_t b) {
+  SCALAR_BOUND(b, HALF_Q, "fqmul input");
+
+  int16_t res = montgomery_reduce((int32_t)a * (int32_t)b);
+
+  SCALAR_BOUND(res, MLKEM_Q, "fqmul output");
+  return res;
+}
```

Comment:
Please add similar comments to this as for `montgomery_reduce_generic`. Move them from reduce.h?

Reply:
Hm, why? Documentation ultimately belongs in the header?

Reply:
Documentation that a client needs to know goes in the .h file. Documentation that an implementer/maintainer needs to know goes in the .c file. No repetition between the two, and no repetition between comments and contracts either.

Reply:
Some repetition is useful, I think, esp. between comments and contracts, given that not everyone knows how to read the latter.

Reply:
The actual content of the contract is a C language expression, which I'd expect every C programmer to be able to read. OK... there's a bit of syntactic sugar and macros to pick up. "For all" remains the only really new concept. I can teach any SDE to read contracts in about half a day...

Reply:
>  I can teach any SDE to read contracts in about half a day...

Sure, but I want any C developer to be able to understand the documentation, without prior teaching. I agree most people will read and interpret the contracts fine, but there's no harm in saying things in plain english as well.